core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -753,10 +761,12 @@ private void acquireSubsetBatchRecords(long requestBaseOffset, long requestLastO
                     .setBaseOffset(offsetState.getKey())
                     .setLastOffset(offsetState.getKey())
                     .setDeliveryCount((short) offsetState.getValue().deliveryCount));
+                atLeastOneAcquired = true;
```

Comment:
We might not need this change if we change the lock acquisition API to accept List<AcquiredRecords> as per the comment above.